### DISCUSSION TITLE
Current state of Llama vs. GPT4All on an M1 Mac in terms of speed?

### DISCUSSION THREAD

**Discussion by gavtography** [abetlen/llama-cpp-python#327] [Q&A]
Just looking for the fastest way to run an LLM on an M1 Mac with Python bindings. Looking for honest opinions on this.

I've already setup my program with GPT4All, but I've heard others saying that there's faster ways on an M1 Mac.

**Comment by ianscrivener** [comment]
`llama-cpp-python` is **VERY** fast with apple Silicon... though must be installed correctly. [MacOS Metal set up instructions](https://bit.ly/3JdHBMu)
